Service

T-Syme opened the pull request:
The service tests have not been finalized - missing the BeforeEach part, but the rest should be complete

T-Syme committed:
Initial code for ServiceDto, ServiceService and ServiceController done

T-Syme committed:
Minor changes in mapping and renamed ServiceTests

T-Syme committed:
Small change to exceptions. Service stuff is set up (no tests)

T-Syme committed:
Merge branch 'main' into service

T-Syme committed:
Service service tests written, but haven't added a "setMockOutput"

T-Syme committed:
Update dependencies in build.gradle

JDziewonski98 reviewed (commented):
looks good to me  so far

sandyl289 commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestServiceService.java:
```diff
@@ -1,4 +1,63 @@
+    }
+
+    @Test // invalid service
+    public void testCreateServiceEmpty() {
```
I think we'll need to add a "testCreateServiceNull" test like in the tutorial notes ( See [Tutorial's complete test suite](https://gist.githubusercontent.com/imbur/d8836dc9f44ece65186a43ac80a259fd/raw/ff59449ad4dfadfc2c54bb317649300f6ae2409c/TestEventRegistrationService.java))
![image](https://user-images.githubusercontent.com/70775435/110049159-492bd680-7d1f-11eb-950f-9e7e8da842d3.png)


Other than that, everything seems good

T-Syme replied:
Done

T-Syme committed:
Added a test for null service name

T-Syme committed:
Modified service tests to use mock output

T-Syme committed:
Modified service tests to use mock output part 2

JDziewonski98 committed:
Merge branch 'main' into service

# Conflicts:
#	RepairShop-Backend/build.gradle

T-Syme committed:
Added a BeforeEach to the service service tests

Also changed createService so that it returns a dto.
Note: commented out a line in technicianService so that it compiles

T-Syme committed:
Merge branch 'main' into service

T-Syme committed:
Uncommented a line in Technician service (it's fine now)

T-Syme commented:
Most changes are complete. Will open a new pull request to merge to main
